src/community/mapml/src/main/java/org/geoserver/mapml/MapMLController.java

```diff
@@ -82,9 +219,44 @@ public Mapml mapML(
             @RequestParam("format") Optional<String> format)
             throws NoSuchAuthorityCodeException, TransformException, FactoryException, IOException {
         LayerInfo layerInfo = geoServer.getCatalog().getLayerByName(layer);
-        if (layerInfo == null) {
-            // TODO better error handling
-            throw new RuntimeException("Invalid layer name: " + layer);
+        // TODO Is DefaultGeographicCRS.WGS84 compatible with Web Mercator / EPSG:3857 ?
+        // If I understand the remarks on www.epsg-registry.org on this account,
+        // the answer is yes, but EPSG:3857 sucks and don't use it. sigh.
```

Comment:
Like to remove TODO's if we can but this is just a community module.

Is there any way resolve your unhappiness with DefaultGeographicCRS.WGS84?

Reply:
I'm not unhappy with DefaultGeographicCRS.WGS84, however the epsg registry says 3857 uses "spherical development" of WGS84, which I find to be a bit opaque in the sense that I am wondering if legitimately I can use a geographic bounding box based on DefaultGeographicCRS.WGS84 with the subsequent code: bbox.expandToInclude(li.getResource().getLatLonBoundingBox()) wherein I'm getting lat-lon bounding boxes and adding them to the default one, but maybe those bounding boxes are incompatible, so that's my main concern.   The TODO should really read "Learn Geotools".

Reply:
To answer your implied question you can use JTS transform method to transform bounds into the same reference system before using expand to include. 

If you want to get extra fancy you can ask it to densify the edges, transform, and then make a new bounds (for bounding boxes near the poles with a curve to them).

See [JTS.transform](https://docs.geotools.org/latest/userguide/library/main/jts.html#transform) in the docs.